File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(selfeigh, height: List[int]) -> int:
        l=0
        r=len(height)-1
        ms=0
        while(l<r):
            width=r-l
            heights=min(height[l],height[r])
            area=heights*width
            if(area>=ms):
                ms=area
            if(height[l]<height[r]):
                l+=1
            else:
                r-=1
        return ms
        

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int l=0;
        int r=height.length-1;
        int ms=0;
        int cs;
        for(int i=0;i<height.length;i++)
        {
            int w=r-l;
            cs=Math.min(height[r],height[l])*w;
            ms=Math.max(cs,ms);

            if(height[l]<height[r])
            {
                l++;
            }
            else
            {
                r--;
            }

        }
        return ms;
        
    }
}